src/model_theory/syntax.lean

```diff
@@ -684,5 +684,32 @@ bounded_formula.is_atomic.equal _ _
 
 end formula
 
+namespace relations
+
+variable (r : L.relations 2)
+
+/-- The sentence indicating that a basic relation symbol is reflexive. -/
+protected def reflexive : L.sentence := ∀' r.bounded_formula₂ &0 &0
+
+/-- The sentence indicating that a basic relation symbol is irreflexive. -/
+protected def irreflexive : L.sentence := ∀' ∼ (r.bounded_formula₂ &0 &0)
+
+/-- The sentence indicating that a basic relation symbol is symmetric. -/
+protected def symmetric : L.sentence := ∀' ∀' (r.bounded_formula₂ &0 &1 ⟹ r.bounded_formula₂ &1 &0)
+
+/-- The sentence indicating that a basic relation symbol is antisymmetric. -/
+protected def antisymmetric : L.sentence :=
+  ∀' ∀' (r.bounded_formula₂ &0 &1 ⟹ (r.bounded_formula₂ &1 &0 ⟹ term.bd_equal &0 &1))
+
+/-- The sentence indicating that a basic relation symbol is transitive. -/
+protected def transitive : L.sentence :=
+  ∀' ∀' ∀' (r.bounded_formula₂ &0 &1 ⟹ (r.bounded_formula₂ &1 &2 ⟹ r.bounded_formula₂ &0 &2))
```

Comment:
if the parentheses around the second implication are necessary, you need to change the way the notation associates: `infix` vs `infixr`